### PR TITLE
Use jsonld lib for import

### LINKED ISSUE #25: Switch to jsonld.js to load and dump documents
The current code has some strong restrictions on what it accepts as input, by treating it as "dumb JSON" instead of processing it like a JSON-LD document. Additionally, it means we need to always handle cases of single-value + list.

Instead, we should use jsonld.js to expand documents, then work on the expanded document which has a more regular format.

#24 also introduces some complexity due to needing to make the output vary depending on the codemeta version (`codemeta:continuousIntegration` + `contIntegration` + `codemeta:isSourceCodeOf` in v2 vs `continuousIntegration` + `codemeta:contIntegration` `isSourceCodeOf` in v3), which would be taken care of by simply using jsonld.js to compact with the right version's context before printing the document.

### PR DESCRIPTION
**Pull request by hjonin**
Resolves #25

### CODE REVIEW

File: js/validation/index.js [hunk 1]
```diff
@@ -100,5 +100,7 @@ function parseAndValidateCodemeta(showPopup) {
         }
     }
 
+    const expanded = await jsonld.expand(parsed);
+    doc = await jsonld.compact(expanded, CODEMETA_CONTEXT_URL);
```
Comment (progval) [inline js/validation/index.js:104]:
```suggestion
    doc = await jsonld.compact(parsed, CODEMETA_CONTEXT_URL);
```

this should be equivalent

**Review by progval — COMMENTED** [review]
Could you add a test to check that we actually support new forms of the same documents? For example, that we can now load documents in fully expanded form.

File: js/validation/index.js (hunk 1 above)
Comment (hjonin) [reply js/validation/index.js:104]: It is indeed!

**Comment by hjonin** [comment]
WIP fix expanded URLs in the test

**Comment by hjonin** [comment]
Ignore previous message, URL was correct

**Comment by progval** [comment]
Is this PR ready for merge?

**Comment by hjonin** [comment]
Yes, sorry I forgot to precise!

**Pull request merged**